generator/src/main/java/com/faforever/neroxis/generator/style/StyleGenerator.java

```diff
@@ -123,12 +123,14 @@ public SCMap generate(GeneratorParameters generatorParameters, long seed) {
         CompletableFuture<Void> heightMapFuture = CompletableFuture.runAsync(terrainGenerator::setHeightmapImage);
         CompletableFuture<Void> textureFuture = CompletableFuture.runAsync(textureGenerator::setTextures);
         CompletableFuture<Void> normalFuture = CompletableFuture.runAsync(textureGenerator::setCompressedDecals);
-        CompletableFuture<Void> previewFuture = CompletableFuture.runAsync(textureGenerator::generatePreview);
+
         CompletableFuture<Void> resourcesFuture = CompletableFuture.runAsync(resourceGenerator::placeResources);
         CompletableFuture<Void> decalsFuture = CompletableFuture.runAsync(decalGenerator::placeDecals);
         CompletableFuture<Void> propsFuture = resourcesFuture.thenAccept(aVoid -> propGenerator.placeProps());
         CompletableFuture<Void> unitsFuture = resourcesFuture.thenAccept(aVoid -> propGenerator.placeUnits());
 
+        CompletableFuture<Void> previewFuture = propsFuture.thenAccept(aVoid -> textureGenerator.generatePreview());
+
```

Comment:
So this change prolly deserves explanation.
Due to the new code in 'PreviewGenerator.addReclaimProps()' where it loops over 'map.getProps()'
It basically needs the propsFuture to complete before starting generatePreview()

So I checked the performance impact and from what I can tell it doesn't impact overall performance much. 

This was a worst case map:
neroxis_map_generator_snapshot_rb5zl3islek2s_aiiaeaq
![image](https://github.com/user-attachments/assets/cdfb1997-b320-4902-b44a-120cf716105c)
Done setHeightMap: 22.56 ms, not found
Done generateResources: 1200.35 ms, not found
Done setCompressedDecals: 795.46 ms, not found
Done generateTextures: 1011.21 ms, not found
Done placeDecals: 1657.39 ms, not found
Done placeProps: 1579.68 ms, not found
Done generatePreview: 163.44 ms, not found
Done setPlacements: 2.55 ms, not found
Pipeline completed!
Map generation done: 5883 ms

This is a map with much less boulders:
neroxis_map_generator_snapshot_px5x6namtp7j2_aiiaeey
![image](https://github.com/user-attachments/assets/5f7b7dd1-451c-4301-abea-4ea4f2b5d946)
Done setHeightMap: 24.08 ms, not found
Done generateResources: 782.38 ms, not found
Done setCompressedDecals: 649.94 ms, not found
Done generateTextures: 856.91 ms, not found
Done placeDecals: 1356.56 ms, not found
Done placeProps: 1433.05 ms, not found
Done generatePreview: 144.63 ms, not found
Done setPlacements: 2.82 ms, not found
Pipeline completed!
Map generation done: 5022 ms